File: p008.py
from operator import mul

def p008(data):
    digits = [int(c) for line in data for c in line if c.isdigit()]
    return max(reduce(mul, digits[i:i + 5]) for i in xrange(len(digits) - 5))

if __name__ == "__main__":
    from benchmark import benchmark
    benchmark(p008, open("data/p008.txt"))

File: p008.clj
(use '[euler :only [run]])

(defn p008 [file]
  (let [content    (slurp file)
        nstr       (clojure.string/replace content #"[^0-9]" "")
        partitions (map #(subs nstr % (+ % 5)) (range (- (count nstr) 4)))
        make-prod  (fn [partition]
                     (reduce * (map #(- % (int \0)) (map int partition))))
        prods      (map make-prod partitions)]
    (apply max prods)))

(run (p008 "data/p008.txt"))
